Doc/library/asyncio-task.rst

```diff
@@ -539,11 +542,23 @@ Shielding From Cancellation
    the ``shield()`` function should be combined with a try/except
    clause, as follows::
 
+       task = asyncio.create_task(something())
+       background_tasks.add(task)
+       task.add_done_callback(background_tasks.discard)
        try:
-           res = await shield(something())
+           res = await shield(task)
        except CancelledError:
            res = None
 
+   .. important::
+
+      Save a reference to tasks passed to this function, to avoid
+      a task disappearing mid execution. The event loop only keeps
+      weak references to tasks. A task that isn't referenced elsewhere
+      may get garbage-collected at any time, even before it's done.
+      For reliable "fire-and-forget" background tasks, gather them in
+      a collection.
```

Comment:
```suggestion
      Save a reference to tasks passed to this function, to avoid
      a task disappearing mid-execution. The event loop only keeps
      weak references to tasks. A task that isn't referenced elsewhere
      may get garbage collected at any time, even before it's done.
      For reliable "fire-and-forget" background tasks, gather them in
      a collection.
```

Reply:
This PR copies the documentation from `asyncio.create_task` and so inherits the hyphenation style.

there's already one use of "mid execution" in cpython and 1 use of "mid-execution" (that's asyncio.create_task where I copied this from)
there's 54 uses of "garbage collected"  and only 25 uses of "garbage-collected"

Reply:
@Akasurde do you want me to change all of these?


```
 graingert@superjacent  testing310  ~/projects/cpython   shield  git grep "mid-execution"
 ✘  graingert@superjacent  testing310  ~/projects/cpython   shield  git grep "mid execution"
Doc/library/asyncio-future.rst:      a task disappearing mid execution.
Doc/library/asyncio-task.rst:      a task disappearing mid execution. The event loop only keeps
Doc/library/asyncio-task.rst:      a task disappearing mid execution. The event loop only keeps
```

Reply:
Yes, sure.